File: src/my_project/easy_problems/from201to250/defuse_the_bomb.py
from typing import List 

class Solution:
    def decrypt(self, code: List[int], k: int) -> List[int]:
        len_code: int = len(code)
        solution: List[int] = list()
        solution = [ 0 for i in range(len_code)]
        count = 0
        if k > 0:
            for i in range(len_code):
                print(i)
                count = 0
                for l in range(i+1, i+1+k):
                    count += code[l%len_code]
                solution[i] = count 
        elif k < 0: 
            for i in range(len_code):
                count = 0
                for l in range(i-1, i-1+k,-1):
                    print(l)
                    count += code[l%len_code]
                solution[i] = count 
        else: 
            return solution 



        return solution

solution = Solution()

print(solution.decrypt(code = [5,7,1,4], k = 3))

print(solution.decrypt(code = [2,4,9,3], k = -2))

File: my_project_typescript/easy_problems_typescript/from101to150/defuseTheBomb.ts
function decrypt(code: number[], k: number): number[] {
    let lenCode: number = code.length
    let solution: Array<number> =  []
    let count: number = 0
    code.map((num, index) => {
        solution.push(0)
    })
    if (k > 0){
        for (let i = 0; i < lenCode; i++){
            count = 0
            for (let l = i+1; l < i+1+k; l++ ){
                count += code[l%lenCode]
            }
            solution[i] = count 
        }
    }
    else if (k < 0){
        for (let i = 0; i < lenCode; i++){
            count = 0
            for (let l = i-1; l > i-1+k; l-- ){
                
                if (l%lenCode <0){
                    count += code[lenCode + (l%lenCode)]
                }
                else {
                    count += code[l%lenCode]
                }
                
            }
            solution[i] = count 
        }
    }
    return solution 
};


console.log(decrypt( [5,7,1,4], 3))

console.log(decrypt( [1,2,3,4], 0))

console.log(decrypt([2,4,9,3], -2))
